src/app/actions/SearchActions.js

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import { Actions } from './Actions';
+import appConfig from '../../../appConfig';
+
+export const searchResults = (results) => {
+  return {
+    type: Actions.SEARCH,
+    results,
+  };
+};
+
+export const search = (query, filter = 'q') => {
+  // Need a parsed query input to use for each filter
+  const userQuery = (query) ? encodeURI(query) : 'test';
+  // Need a client to send the search and receive results
+  // Need to pass the results to a renderer
+  const apiUrl = appConfig.esUrl.development;
+
+  return (dispatch) => {
+    return axios.get(apiUrl, { params: { q: userQuery } })
```

Comment:
You mentioned that in your function you are not returning a promise like I am in my branch, but this axios call is a promise so they're both alike.

Reply:
Right -- I was just wondering if use of `new Promise(resolve, reject)` with an internal call to the search function separately was a better approach than just calling axios here.

Reply:
Leave this as is and refactor later if you find that's helpful.